synapse/storage/data_stores/main/registration.py

```diff
@@ -503,9 +498,11 @@ def _find_next_generated_user_id(txn):
                 match = regex.search(user_id)
                 if match:
                     found.add(int(match.group(1)))
```

Comment:
can we avoid building a set if we're jsut going to use one value? track the most you've found so far.

(the set is pretty big)

Reply:
I guess, though I was hoping to get a minimal fix out, but can try and shuffle